src/Security/SudoMode/SudoModeService.php

```diff
@@ -45,6 +49,11 @@ public function activate(Session $session): bool
         return true;
     }
 
+    public function deactivate(Session $session): void
+    {
+        $session->set(SudoModeService::SUDO_MODE_SESSION_KEY, null);
+    }
```

Comment:
What is this new method for? After it's activated, shouldn't it remain active for the full request cycle?

Reply:
When logging in with a fresh member for the first time (possibly other scenarios too), if the MFA module is installed you'll be prompted to sign up for MFA on the frontend. If the user does so, then MFA will be silently activated so that the sudo checks in controller endpoints pass.  

This still needs to happen, though as we're removing the 'automatically activate sudo mode when logging in' bit, we need to deactivate the silently activated sudo mode.

Reply:
MFA being activated shouldn't disable/activate/bypass sudo mode, should it?

> This still needs to happen

Why? Sounds a bit pointless if we're immediately deactivating it anyway?

If there's something in MFA that requires MFA to be active to set it up, sounds like the MFA module should be updated so that doesn't happen if being set from that front-end prompt.

Reply:
This discussion will continue in https://github.com/silverstripe/silverstripe-mfa/pull/587#discussion_r1955289016 which is where the MFA changes need to happen